lisa/tests/base.py

```diff
@@ -466,6 +469,75 @@ def test_noisy_tasks(self, noise_threshold_pct=None, noise_threshold_ms=None):
         res.add_metric("noisiest task", metric)
 
         return res
+
+    @classmethod
+    #pylint: disable=unused-argument
+    def check_noisy_tasks(cls, noise_threshold_pct=None, noise_threshold_ms=None):
+        """
+        Decorator that applies :meth:`test_noisy_tasks` to the underlying function
+
+        We also expose :meth:`test_noisy_tasks` parameters to the decorated
+        function.
+        """
+        def decorator(func):
+            @cls.test_noisy_tasks.used_events
+            @functools.wraps(func)
+            def wrapper(self, *args,
+                        noise_threshold_pct=noise_threshold_pct,
+                        noise_threshold_ms=noise_threshold_ms,
+                        **kwargs):
+                res = func(self, *args, **kwargs)
+
+                print(noise_threshold_pct)
+
+                noise_res = self.test_noisy_tasks(
+                    noise_threshold_pct, noise_threshold_ms)
+                res.metrics.update(noise_res.metrics)
+
+                if not noise_res:
+                    res.result = Result.UNDECIDED
+
+                return res
+
+            # https://stackoverflow.com/a/33112180
+            # The wrapper has all of `func`'s parameters plus `test_noisy_tasks`',
+            # but since we use `wraps(func)` we'll only get doc/autocompletion for
+            # `func`'s. Expose the extra parameters to the decorated function to
+            # make it more user friendly.
+            func_sig = signature(func)
+            dec_params = signature(cls.check_noisy_tasks).parameters
+            new_params = []
+            # Propagate the default values given to the decorator
+            for name, param in dec_params.items():
+                new_params.append(param.replace(default=locals()[name]))
+
+            wrapper.__signature__ = func_sig.replace(
+                parameters=list(func_sig.parameters.values()) + new_params
+            )
+
+            # Make it obvious in the doc where the extra parameters come from
+            merged_doc = cls.test_noisy_tasks.__doc__.splitlines()
+            indent = " " * (len(merged_doc[1]) - len(merged_doc[1].lstrip()))
+
+            separator = textwrap.dedent(
+                """
+                **Added by** :meth:`~{}.{}.{}`:
+
+                The returned ``ResultBundle.result`` will be changed to
+                :attr:`~lisa.tests.base.Result.UNDECIDED` if the environment was
+                too noisy:
+                """.format(cls.__module__, cls.__name__, cls.check_noisy_tasks.__name__)
+            )
+            separator = textwrap.indent(separator, indent)
```

Comment:
do we really need to re-indent ? Generated docs will look the same,  and the user will probably not care a lot as long as the first column used is consistent. That would simplify into:
```
doc = textwrap.dedent(cls.test_noisy_tasks.__doc__)
separator = textwrap.dedent(...)
doc = [separator] + doc.splitlines()[1:]
wrapper.__doc__ = textwrap.dedent(wrapper_doc) + '\n' + '\n'.join(doc)
```

The last `textwrap.dedent` is necessary since it cannot be assumed that the docstring of the wrapper has the same indendation than the one we just created, so we dedent everything to ensure proper alignment.

Reply:
From what I've experienced Sphinx/rst go out of whack where there are mixed indentation levels. The text is still readable, but things like formating for parameters gets broken :(

Reply:
I experienced that as well, that's why I came up with always dedenting everything, so there is no mixed indentation in the output. The solution above dedents every string it uses before stitching them together for that reason.

Reply:
Right, that's the other way around but it is simpler, I'll go with that.